Crash on 1.21.7

The server simply shut down throwing this error, tested on a fresh Paper install, 1.21.7 (last dev build).


```[23:27:46 ERROR]: [ChunkTaskScheduler] Chunk system error at chunk (12,-17), holder: NewChunkHolder{world=mundo, chunkX=12, chunkZ=-17, entityChunkFromDisk=false, lastChunkCompletion={chunk_class=net.minecraft.world.level.chunk.ProtoChunk,status=minecraft:carvers}, currentGenStatus=minecraft:carvers, requestedGenStatus=minecraft:full, generationTask=ChunkProgressionTask{class: ca.spottedleaf.moonrise.patches.chunk_system.scheduling.task.ChunkUpgradeGenericStatusTask, for world: mundo, chunk: (12,-17), hashcode: 269733263, priority: COMPLETING, status: minecraft:features, scheduled: true}, generationTaskStatus=minecraft:features, priority=null, priorityLocked=false, neighbourRequestedPriority=null, effective_priority=null, oldTicketLevel=33, currentTicketLevel=33, totalNeighboursUsingThisChunk=17, fullNeighbourChunksLoadedBitset=32505856, currentChunkStatus=INACCESSIBLE, pendingChunkStatus=INACCESSIBLE, is_unload_safe=ticket_level, killed=false}, exception:
java.lang.Throwable: java.lang.NoSuchMethodError: 'net.minecraft.world.level.LevelAccessor de.derfrzocker.custom.ore.generator.impl.v1_21_R5.ChunkAccessImpl.getWorld()'
        at ca.spottedleaf.moonrise.patches.chunk_system.scheduling.ChunkTaskScheduler.unrecoverableChunkSystemFailure(ChunkTaskScheduler.java:314) ~[purpur-1.21.7.jar:1.21.7-2475-c4e5604]
        at ca.spottedleaf.moonrise.patches.chunk_system.scheduling.NewChunkHolder.lambda$setGenerationTask$5(NewChunkHolder.java:1629) ~[purpur-1.21.7.jar:1.21.7-2475-c4e5604]
        at ca.spottedleaf.moonrise.patches.chunk_system.scheduling.task.ChunkProgressionTask.complete0(ChunkProgressionTask.java:91) ~[purpur-1.21.7.jar:1.21.7-2475-c4e5604]
        at ca.spottedleaf.moonrise.patches.chunk_system.scheduling.task.ChunkProgressionTask.complete(ChunkProgressionTask.java:74) ~[purpur-1.21.7.jar:1.21.7-2475-c4e5604]
        at ca.spottedleaf.moonrise.patches.chunk_system.scheduling.task.ChunkUpgradeGenericStatusTask.run(ChunkUpgradeGenericStatusTask.java:116) ~[purpur-1.21.7.jar:1.21.7-2475-c4e5604]
        at ca.spottedleaf.moonrise.patches.chunk_system.scheduling.executor.RadiusAwarePrioritisedExecutor$Task.run(RadiusAwarePrioritisedExecutor.java:492) ~[purpur-1.21.7.jar:1.21.7-2475-c4e5604]
        at ca.spottedleaf.concurrentutil.executor.queue.PrioritisedTaskQueue$PrioritisedQueuedTask.execute(PrioritisedTaskQueue.java:281) ~[concurrentutil-0.0.3.jar:?]
        at ca.spottedleaf.concurrentutil.executor.queue.PrioritisedTaskQueue.executeTask(PrioritisedTaskQueue.java:101) ~[concurrentutil-0.0.3.jar:?]
        at ca.spottedleaf.concurrentutil.executor.thread.PrioritisedThreadPool$ExecutorGroup$ThreadPoolExecutor.executeTask(PrioritisedThreadPool.java:533) ~[concurrentutil-0.0.3.jar:?]
        at ca.spottedleaf.concurrentutil.executor.thread.PrioritisedThreadPool$PrioritisedThread.pollTasks(PrioritisedThreadPool.java:354) ~[concurrentutil-0.0.3.jar:?]
        at ca.spottedleaf.concurrentutil.executor.thread.PrioritisedQueueExecutorThread.doRun(PrioritisedQueueExecutorThread.java:60) ~[concurrentutil-0.0.3.jar:?]
        at ca.spottedleaf.concurrentutil.executor.thread.PrioritisedQueueExecutorThread.run(PrioritisedQueueExecutorThread.java:49) ~[concurrentutil-0.0.3.jar:?]
Caused by: java.lang.NoSuchMethodError: 'net.minecraft.world.level.LevelAccessor de.derfrzocker.custom.ore.generator.impl.v1_21_R5.ChunkAccessImpl.getWorld()'
        at custom-ore-generator-2025.07.13.jar/de.derfrzocker.custom.ore.generator.impl.v1_21_R5.ChunkAccessImpl.getHeight(ChunkAccessImpl.java:65) ~[custom-ore-generator-2025.07.13.jar:?]
        at net.minecraft.world.level.levelgen.feature.OreFeature.place(OreFeature.java:46) ~[purpur-1.21.7.jar:1.21.7-2475-c4e5604]
        at custom-ore-generator-2025.07.13.jar/de.derfrzocker.custom.ore.generator.impl.v1_21_R5.oregenerator.MinableGenerator_v1_21_R5.generate(MinableGenerator_v1_21_R5.java:96) ~[custom-ore-generator-2025.07.13.jar:?]
        at custom-ore-generator-2025.07.13.jar/de.derfrzocker.custom.ore.generator.impl.v1_21_R5.CustomOrePopulator.generate(CustomOrePopulator.java:141) ~[custom-ore-generator-2025.07.13.jar:?]
        at custom-ore-generator-2025.07.13.jar/de.derfrzocker.custom.ore.generator.impl.v1_21_R5.CustomOrePopulator.lambda$populate$2(CustomOrePopulator.java:91) ~[custom-ore-generator-2025.07.13.jar:?]
        at java.base/java.util.Arrays$ArrayList.forEach(Arrays.java:4263) ~[?:?]
        at custom-ore-generator-2025.07.13.jar/de.derfrzocker.custom.ore.generator.impl.v1_21_R5.CustomOrePopulator.lambda$populate$3(CustomOrePopulator.java:91) ~[custom-ore-generator-2025.07.13.jar:?]
        at java.base/java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
        at custom-ore-generator-2025.07.13.jar/de.derfrzocker.custom.ore.generator.impl.v1_21_R5.CustomOrePopulator.populate(CustomOrePopulator.java:88) ~[custom-ore-generator-2025.07.13.jar:?]
        at net.minecraft.world.level.chunk.ChunkGenerator.applyBiomeDecoration(ChunkGenerator.java:456) ~[purpur-1.21.7.jar:1.21.7-2475-c4e5604]
        at net.minecraft.world.level.chunk.ChunkGenerator.applyBiomeDecoration(ChunkGenerator.java:439) ~[purpur-1.21.7.jar:1.21.7-2475-c4e5604]
        at net.minecraft.world.level.chunk.status.ChunkStatusTasks.generateFeatures(ChunkStatusTasks.java:150) ~[purpur-1.21.7.jar:1.21.7-2475-c4e5604]
        at net.minecraft.world.level.chunk.status.ChunkStep.apply(ChunkStep.java:66) ~[purpur-1.21.7.jar:1.21.7-2475-c4e5604]
        at ca.spottedleaf.moonrise.patches.chunk_system.scheduling.task.ChunkUpgradeGenericStatusTask.run(ChunkUpgradeGenericStatusTask.java:99) ~[purpur-1.21.7.jar:1.21.7-2475-c4e5604]
        ... 7 more
```

--- a/impl/v1_21_R7/src/main/java/de/derfrzocker/custom/ore/generator/impl/v1_21_R7/ChunkAccessImpl.java
+++ b/impl/v1_21_R7/src/main/java/de/derfrzocker/custom/ore/generator/impl/v1_21_R7/ChunkAccessImpl.java
@@ -18,16 +18,25 @@ public class ChunkAccessImpl
         extends BlockStateListPopulator
         implements ChunkAccess {
 
+    private final LevelAccessor world;
     private final Set<BlockPos> blockPosSet = new HashSet<>();
     private final Consumer<BlockPos> blockSet = blockPosSet::add;
 
     public ChunkAccessImpl(LevelAccessor world) {
         super(world);
+        this.world = world;
+    }
+    
+    // In Paper 1.21.11, getWorld() was removed from BlockStateListPopulator
+    public LevelAccessor getWorld() {
+        return world;
     }
 
     @Override
     public void setMaterial(@NotNull Material material, int x, int y, int z) {
-        setBlock(new BlockPos(x, y, z), ((CraftBlockData) material.createBlockData()).getState(), 3);
+        BlockPos pos = new BlockPos(x, y, z);
+        setBlock(pos, ((CraftBlockData) material.createBlockData()).getState(), 3);
+        blockPosSet.add(pos);
     }
 
     @NotNull
@@ -47,9 +56,9 @@ public class ChunkAccessImpl
 
     @Override
     public Set<BlockPos> getBlocks() {
-        Set<BlockPos> blockPos = new HashSet<>(super.getBlocks());
-        blockPos.addAll(blockPosSet);
-        return blockPos;
+        // In Paper 1.21.11, getBlocks() was removed from BlockStateListPopulator
+        // We track blocks ourselves in blockPosSet (populated in setMaterial)
+        return new HashSet<>(blockPosSet);
     }
 
     @Override

--- a/impl/v1_21_R7/src/main/java/de/derfrzocker/custom/ore/generator/impl/v1_21_R7/CustomOrePopulator.java
+++ b/impl/v1_21_R7/src/main/java/de/derfrzocker/custom/ore/generator/impl/v1_21_R7/CustomOrePopulator.java
@@ -114,8 +114,8 @@ public class CustomOrePopulator extends BlockPopulator {
 
         ChunkAccessImpl chunkAccess = new ChunkAccessImpl(((CraftLimitedRegion) limitedRegion).getHandle());
         oreGenerator.generate(oreConfig, chunkAccess, chunkX, chunkZ, random, biome, biomeLocations);
-        chunkAccess.refreshTiles();
-        chunkAccess.updateList();
+        // refreshTiles() and updateList() methods were removed in Paper 1.21.11
+        // Tile entities are now handled automatically by the new BlockStateListPopulator API
         for (BlockPos pos : chunkAccess.getBlocks()) {
             oreConfig.getCustomData().forEach((customData, object) -> customData.getCustomDataApplier().apply(oreConfig, new Location(null, pos.getX(), pos.getY(), pos.getZ()), limitedRegion));
         }

--- a/impl/v26_1_base/src/main/java/de/derfrzocker/custom/ore/generator/impl/v26_1_base/ChunkAccessImpl.java
+++ b/impl/v26_1_base/src/main/java/de/derfrzocker/custom/ore/generator/impl/v26_1_base/ChunkAccessImpl.java
@@ -16,16 +16,25 @@ import org.jetbrains.annotations.NotNull;
 
 public class ChunkAccessImpl extends BlockStateListPopulator implements ChunkAccess {
 
+    private final LevelAccessor world;
     private final Set<BlockPos> blockPosSet = new HashSet<>();
     private final Consumer<BlockPos> blockSet = blockPosSet::add;
 
     public ChunkAccessImpl(LevelAccessor world) {
         super(world);
+        this.world = world;
+    }
+    
+    // In Paper 1.21.11, getWorld() was removed from BlockStateListPopulator
+    public LevelAccessor getWorld() {
+        return world;
     }
 
     @Override
     public void setMaterial(@NotNull Material material, int x, int y, int z) {
-        setBlock(new BlockPos(x, y, z), ((CraftBlockData) material.createBlockData()).getState(), 3);
+        BlockPos pos = new BlockPos(x, y, z);
+        setBlock(pos, ((CraftBlockData) material.createBlockData()).getState(), 3);
+        blockPosSet.add(pos);
     }
 
     @NotNull
@@ -45,9 +54,9 @@ public class ChunkAccessImpl extends BlockStateListPopulator implements ChunkAcc
 
     @Override
     public Set<BlockPos> getBlocks() {
-        Set<BlockPos> blockPos = new HashSet<>(super.getBlocks());
-        blockPos.addAll(blockPosSet);
-        return blockPos;
+        // In Paper 1.21.11, getBlocks() was removed from BlockStateListPopulator
+        // We track blocks ourselves in blockPosSet (populated in setMaterial)
+        return new HashSet<>(blockPosSet);
     }
 
     @Override

--- a/impl/v26_1_base/src/main/java/de/derfrzocker/custom/ore/generator/impl/v26_1_base/CustomOrePopulator.java
+++ b/impl/v26_1_base/src/main/java/de/derfrzocker/custom/ore/generator/impl/v26_1_base/CustomOrePopulator.java
@@ -166,8 +166,8 @@ public class CustomOrePopulator extends BlockPopulator {
 
         ChunkAccessImpl chunkAccess = new ChunkAccessImpl(((CraftLimitedRegion) limitedRegion).getHandle());
         oreGenerator.generate(oreConfig, chunkAccess, chunkX, chunkZ, random, biome, biomeLocations);
-        chunkAccess.refreshTiles();
-        chunkAccess.updateList();
+        // refreshTiles() and updateList() methods were removed in Paper 1.21.11
+        // Tile entities are now handled automatically by the new BlockStateListPopulator API
         for (BlockPos pos : chunkAccess.getBlocks()) {
             oreConfig
                     .getCustomData()

